default_warehouse_from_sale_team/models/account_move.py

```diff
@@ -7,20 +7,18 @@ class AccountMove(models.Model):
     @api.model
     def _search_default_journal(self, journal_types):
         """If a team is provided and it has a sales journal set, take it as 1st alternative"""
-        team = self.env.context.get("salesteam")
-        journal_on_team = None
-        if team and journal_types == ["sale"]:
-            journal_on_team = team._get_default_journal_sale()
+        team = self.env.context.get("salesteam", self.team_id)
+        journal_on_team = team._get_default_journal(journal_types)
```

Comment:
you could use 

is_sale_document/ is_purchase_document

what happen with moves created manually 

and credit notes?

Reply:
- Credit notes should also be considered
- In case of journal entries, it won't be triggered by the onchange because the onchange validates it's an invoice `is_invoice` which is `is_sale_document` + `is_purchase_document`, so the only time this could be called is when created, when `self.team_id` is not set. I could validate it, though.

Reply:
FYI I double-checked this doesn't affect journal entries and I can confirm it does not, so resolving.